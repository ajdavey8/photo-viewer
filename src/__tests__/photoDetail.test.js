import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PhotoDetail from '../components/photoDetail';

describe(PhotoDetail, () => {
  const file = { "file": "1 Alan Berryhill 2018-04-17 person, outdoors, rain.jpg"};
  const component = shallow(<PhotoDetail photo={file} />);

  it('renders and matches snapshot', () => {
    const tree = renderer
      .create(<PhotoDetail photo={file} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the Name of the photographer', () => {
    const name = component.find('h2');
    expect(name.contains("Alan Berryhill")).toEqual(true);
  });
});
