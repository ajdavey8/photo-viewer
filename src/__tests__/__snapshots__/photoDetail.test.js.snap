// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class PhotoDetail extends _react.Component {

  photographer(file) {
    const name = file.split(" ");
    return name[1] + " " + name[2];
  }

  date(file) {
    const date = file.split(" ");
    return date[3].split('-').reverse().join('.');
  }

  tags(file) {
    const shortenedfile = file.replace(/.jpg/g, '');
    const tags = shortenedfile.split(" ");
    tags.splice(0, 4);

    return tags.join(" ");
  }

  render() {
    return (
      _react2.default.createElement('div', { className: 'legend', __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this },
        _react2.default.createElement('h3', { __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, ' ', this.date(this.props.photo.file)),
        _react2.default.createElement('h2', { __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }, ' ', this.photographer(this.props.photo.file)),
        _react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this }, ' Tags: ', this.tags(this.props.photo.file))));


  }} renders and matches snapshot 1`] = `
<div
  className="legend"
>
  <h3>
     
    17.04.2018
  </h3>
  <h2>
     
    Alan Berryhill
  </h2>
  <p>
     Tags: 
    person, outdoors, rain
  </p>
</div>
`;
